plugins/modules/aci_epg_useg_attribute.py

```diff
@@ -0,0 +1,426 @@
+        criterion=dict(type="list", elements="str"),  # Criterion list
+        attribute=dict(type="str", aliases=["attribute_name"]),
+        type=dict(type="str", choices=list(ATTRIBUTE_MAPPING.keys())),
+        operator=dict(type="str", choices=["equals", "contains", "starts_with", "ends_with"]),
+        category=dict(type="str", aliases=["custom_attribute"]),
+        value=dict(type="str"),
+        use_subnet=dict(type="str", choices=["yes", "no"])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["ap", "epg", "tenant", "attribute", "type"]],
+            ["state", "present", ["ap", "epg", "tenant", "attribute", "type"]]
+        ]
+    )
+
+    aci = ACIModule(module)
+
+    ap = module.params.get("ap")
+    epg = module.params.get("epg")
+    state = module.params.get("state")
+    tenant = module.params.get("tenant")
+    criterion = module.params.get("criterion")
+    attr_name = module.params.get("attribute")
+    type = module.params.get("type")
+    attr_value = module.params.get("value")
+    attr_operator = module.params.get("operator")
+    attr_category = module.params.get("category")
+    attr_use_subnet = module.params.get("use_subnet")
+
+    #  attribute class and config
+    attr_class = ATTRIBUTE_MAPPING[type]["attr_class"]
+    attr_type = ATTRIBUTE_MAPPING[type]["attr_type"]
+    attr_config = dict(name=attr_name)
+
+    if attr_class == "fvVmAttr":
+        attr_rn = "vmattr-{0}".format(attr_name)
+        attr_config.update(type=attr_type)
+        if attr_value is not None:
+            attr_config.update(value=attr_value)
+        if attr_operator is not None:
+            attr_config.update(operator=OPERATOR_MAPPING[attr_operator])
+
+        if type == "vm_custom_attr":
+            if attr_category is not None:
+                attr_config.update(labelName=attr_category)
+        elif type == "vm_tag":
+            if attr_category is not None:
+                attr_config.update(category=attr_category)
+
+    elif attr_class == "fvIpAttr":
+        attr_rn = "ipattr-{0}".format(attr_name)
+        if attr_use_subnet == "yes":
+            attr_config.update(usefvSubnet=attr_use_subnet)
+        else:
+            if attr_value is not None:
+                attr_config.update(ip=attr_value)
+
+    elif attr_class == "fvMacAttr":
+        attr_rn = "macattr-{0}".format(attr_name)
+        if attr_value is not None:
+            attr_config.update(mac=attr_value.upper())
+
+    #  criterion class and building relative name
+    crtrn_rn = "crtrn"
+    crtrn_class = "fvCrtrn"
+    crtrn_name = "default"
+    if criterion:
+        if len(criterion) > 3:
+            module.fail_json(msg="Depth of sub criterion %s exceeds maximum limit 3." % len(criterion))
+        crtrn_name = criterion[-1]
+        crtrn_class = "fvSCrtrn"
+        crtrn_rn = "crtrn/crtrn-" + "/crtrn-".join(criterion)
```

Comment:
please add test cases that handles the error and the fvSCrtrn functionality

Reply:
I will add a test for exceeding the maximum limit. The implementation of fvSCrtrn class needs a discussion.

Reply:
ok when will you have time for this? if you can propose timeslot via mail we can set up a meeting